python/lsst/obs/fiberspectrograph/translator.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import os
+
+import astropy.units as u
+from astropy.time import Time
+
+from astro_metadata_translator import cache_translation
+from lsst.obs.lsst.translators.lsst import SIMONYI_TELESCOPE, LsstBaseTranslator
+
+from lsst.utils import getPackageDir
+
+__all__ = ["FiberSpectrographTranslator", ]
+
+log = logging.getLogger(__name__)
+
+
+class FiberSpectrographTranslator(LsstBaseTranslator):
+    """Metadata translator for Rubin calibration fibre spectrographs headers"""
+
+    name = "FiberSpectrograph"
+    """Name of this translation class"""
+
+    supported_instrument = "FiberSpec"
+    """Supports the Rubin calibration fibre spectrographs."""
+
+    default_search_path = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default search path to use to locate header correction files."""
+
+    default_resource_root = os.path.join(getPackageDir("obs_fiberspectrograph"), "corrections")
+    """Default resource path root to use to locate header correction files."""
+
+    DETECTOR_MAX = 1
+
+    _const_map = {
+        "detector_num": 0,
+        "detector_name": "0",
+        "exposure_group": None,
+        "object": None,
+        "physical_filter": "empty",
+        "detector_serial": "0xdeadbeef",
+        "detector_group": "None",
+        "relative_humidity": None,
+        "pressure": None,
+        "temperature": None,
+        "focus_z": None,
+        "boresight_airmass": None,
+        "boresight_rotation_angle": None,
+        "tracking_radec": None,
+        "telescope": SIMONYI_TELESCOPE,
+        "observation_type": "spectrum",  # IMGTYPE is ''
+    }
+    """Constant mappings"""
+
+    _trivial_map = {
+        "observation_id": "OBSID",
+        "science_program": ("PROGRAM", dict(default="unknown")),
+    }
+    """One-to-one mappings"""
+
+    @classmethod
+    def can_translate(cls, header, filename=None):
+        """Indicate whether this translation class can translate the
+        supplied header.
+
+        Parameters
+        ----------
+        header : `dict`-like
+            Header to convert to standardized form.
+        filename : `str`, optional
+            Name of file being translated.
+
+        Returns
+        -------
+        can : `bool`
+            `True` if the header is recognized by this class. `False`
+            otherwise.
+        """
+
+        return "INSTRUME" in header and header["INSTRUME"] in ["FiberSpectrograph.Broad"]
```

Comment:
Does `FiberSpectrograph.Broad` imply that there are other instruments that are not broad? If so this class will not distinguish between them in any way since everything will be FiberSpec. Will people need to be able to do butler queries on broad and not broad?

Reply:
Here as well @RobertLuptonTheGood would know

Reply:
I think if you are only going to support one of the fiber spectrographs to begin with you do have to make sure you choose the write name for it when it's ingested into butler.

Reply:
Did this get resolved? ie is the broad going to be FiberSpec and the other variants will get different names when they start to arrive?